File: scripts/importar-biblia-acf.ts
import { createClient } from '@supabase/supabase-js'

const supabaseUrl = process.env.SUPABASE_URL!
const supabaseKey = process.env.SUPABASE_SERVICE_ROLE_KEY!

const supabase = createClient(supabaseUrl, supabaseKey)

// Mapeamento de abreviações para IDs dos livros
const livrosMap: Record<string, number> = {
  'gn': 1, 'ex': 2, 'lv': 3, 'nm': 4, 'dt': 5, 'js': 6, 'jz': 7, 'rt': 8,
  '1sm': 9, '2sm': 10, '1rs': 11, '2rs': 12, '1cr': 13, '2cr': 14, 'ed': 15,
  'ne': 16, 'et': 17, 'job': 18, 'sl': 19, 'pv': 20, 'ec': 21, 'ct': 22,
  'is': 23, 'jr': 24, 'lm': 25, 'ez': 26, 'dn': 27, 'os': 28, 'jl': 29,
  'am': 30, 'ob': 31, 'jn': 32, 'mq': 33, 'na': 34, 'hc': 35, 'sf': 36,
  'ag': 37, 'zc': 38, 'ml': 39, 'mt': 40, 'mc': 41, 'lc': 42, 'jo': 43,
  'at': 44, 'rm': 45, '1co': 46, '2co': 47, 'gl': 48, 'ef': 49, 'fp': 50,
  'cl': 51, '1ts': 52, '2ts': 53, '1tm': 54, '2tm': 55, 'tt': 56, 'fm': 57,
  'hb': 58, 'tg': 59, '1pe': 60, '2pe': 61, '1jo': 62, '2jo': 63, '3jo': 64,
  'jd': 65, 'ap': 66
}

async function buscarCapitulo(abreviacao: string, capitulo: number): Promise<string | null> {
  try {
    const url = `https://www.abibliadigital.com.br/api/verses/acf/${abreviacao}/${capitulo}`
    console.log(`[v0] Buscando: ${url}`)
    
    const response = await fetch(url)
    
    if (!response.ok) {
      console.error(`[v0] Erro HTTP ${response.status} para ${abreviacao} ${capitulo}`)
      return null
    }
    
    const data = await response.json()
    
    if (data.verses && Array.isArray(data.verses)) {
      const texto = data.verses.map((v: any) => v.text).join(' ')
      return texto
    }
    
    return null
  } catch (error) {
    console.error(`[v0] Erro ao buscar ${abreviacao} ${capitulo}:`, error)
    return null
  }
}

async function importarBiblia() {
  console.log('[v0] Iniciando importação da Bíblia ACF...')
  
  // Buscar todos os capítulos vazios
  const { data: capitulos, error } = await supabase
    .from('capitulos_biblia')
    .select('id, livro_id, numero_capitulo, livros_biblia(abreviacao)')
    .is('texto', null)
    .order('livro_id')
    .order('numero_capitulo')
  
  if (error) {
    console.error('[v0] Erro ao buscar capítulos:', error)
    return
  }
  
  console.log(`[v0] Encontrados ${capitulos?.length || 0} capítulos para importar`)
  
  let importados = 0
  let erros = 0
  
  for (const cap of capitulos || []) {
    const abreviacao = (cap.livros_biblia as any)?.abreviacao
    
    if (!abreviacao) {
      console.error(`[v0] Abreviação não encontrada para livro_id ${cap.livro_id}`)
      erros++
      continue
    }
    
    const texto = await buscarCapitulo(abreviacao, cap.numero_capitulo)
    
    if (texto) {
      const { error: updateError } = await supabase
        .from('capitulos_biblia')
        .update({ texto })
        .eq('id', cap.id)
      
      if (updateError) {
        console.error(`[v0] Erro ao atualizar capítulo ${cap.id}:`, updateError)
        erros++
      } else {
        importados++
        console.log(`[v0] Importado: ${abreviacao} ${cap.numero_capitulo} (${importados}/${capitulos.length})`)
      }
    } else {
      erros++
    }
    
    // Aguardar 500ms entre requisições para não sobrecarregar a API
    await new Promise(resolve => setTimeout(resolve, 500))
  }
  
  console.log(`[v0] Importação concluída!`)
  console.log(`[v0] Capítulos importados: ${importados}`)
  console.log(`[v0] Erros: ${erros}`)
}

importarBiblia()

File: scripts/importar-biblia-acf.py
import os
import sys
import requests
import time
from supabase import create_client, Client

try:
    import requests
    from supabase import create_client, Client
except ImportError as e:
    print("❌ Erro: Dependências não instaladas!")
    print("\nPara instalar as dependências necessárias, execute:")
    print("\n  pip install supabase requests")
    print("\nOu se você usa Python 3:")
    print("\n  pip3 install supabase requests")
    sys.exit(1)

# Configuração do Supabase
supabase_url = os.environ.get("SUPABASE_URL")
supabase_key = os.environ.get("SUPABASE_SERVICE_ROLE_KEY")

if not supabase_url or not supabase_key:
    print("❌ Erro: Variáveis de ambiente não configuradas!")
    print("\nConfigure as seguintes variáveis de ambiente:")
    print("  SUPABASE_URL")
    print("  SUPABASE_SERVICE_ROLE_KEY")
    sys.exit(1)

supabase: Client = create_client(supabase_url, supabase_key)

# Mapeamento de nomes para abreviações da API
LIVROS_MAP = {
    "Gênesis": "gn", "Êxodo": "ex", "Levítico": "lv", "Números": "nm", "Deuteronômio": "dt",
    "Josué": "js", "Juízes": "jz", "Rute": "rt", "1 Samuel": "1sm", "2 Samuel": "2sm",
    "1 Reis": "1rs", "2 Reis": "2rs", "1 Crônicas": "1cr", "2 Crônicas": "2cr",
    "Esdras": "ed", "Neemias": "ne", "Ester": "et", "Jó": "job", "Salmos": "sl",
    "Provérbios": "pv", "Eclesiastes": "ec", "Cânticos": "ct", "Isaías": "is",
    "Jeremias": "jr", "Lamentações": "lm", "Ezequiel": "ez", "Daniel": "dn",
    "Oséias": "os", "Joel": "jl", "Amós": "am", "Obadias": "ob", "Jonas": "jn",
    "Miquéias": "mq", "Naum": "na", "Habacuque": "hc", "Sofonias": "sf", "Ageu": "ag",
    "Zacarias": "zc", "Malaquias": "ml",
    "Mateus": "mt", "Marcos": "mc", "Lucas": "lc", "João": "jo", "Atos": "at",
    "Romanos": "rm", "1 Coríntios": "1co", "2 Coríntios": "2co", "Gálatas": "gl",
    "Efésios": "ef", "Filipenses": "fp", "Colossenses": "cl", "1 Tessalonicenses": "1ts",
    "2 Tessalonicenses": "2ts", "1 Timóteo": "1tm", "2 Timóteo": "2tm", "Tito": "tt",
    "Filemom": "fm", "Hebreus": "hb", "Tiago": "tg", "1 Pedro": "1pe", "2 Pedro": "2pe",
    "1 João": "1jo", "2 João": "2jo", "3 João": "3jo", "Judas": "jd", "Apocalipse": "ap"
}

def buscar_livros():
    """Busca todos os livros do banco de dados"""
    response = supabase.table("livros_biblia").select("*").order("ordem").execute()
    return response.data

def buscar_capitulo_api(abreviacao: str, numero_capitulo: int, versao: str = "acf"):
    """Busca o texto de um capítulo da API ABíbliaDigital"""
    url = f"https://www.abibliadigital.com.br/api/verses/{versao}/{abreviacao}/{numero_capitulo}"
    
    try:
        response = requests.get(url, timeout=10)
        if response.status_code == 200:
            data = response.json()
            # Concatenar todos os versículos em um único texto
            verses = data.get("verses", [])
            texto_completo = " ".join([v.get("text", "") for v in verses])
            return texto_completo
        else:
            print(f"❌ Erro ao buscar {abreviacao} {numero_capitulo}: Status {response.status_code}")
            return None
    except Exception as e:
        print(f"❌ Erro ao buscar {abreviacao} {numero_capitulo}: {e}")
        return None

def atualizar_capitulo(livro_id: int, numero_capitulo: int, texto: str):
    """Atualiza o texto de um capítulo no banco"""
    try:
        supabase.table("capitulos_biblia").update({
            "texto": texto
        }).eq("livro_id", livro_id).eq("numero_capitulo", numero_capitulo).execute()
        return True
    except Exception as e:
        print(f"❌ Erro ao salvar capítulo: {e}")
        return False

def importar_biblia():
    """Importa toda a Bíblia ACF"""
    print("🚀 Iniciando importação da Bíblia ACF...")
    
    livros = buscar_livros()
    total_livros = len(livros)
    total_capitulos = 0
    sucesso = 0
    falhas = 0
    
    for idx, livro in enumerate(livros, 1):
        nome = livro["nome"]
        livro_id = livro["id"]
        total_caps = livro["total_capitulos"]
        abreviacao = LIVROS_MAP.get(nome)
        
        if not abreviacao:
            print(f"⚠️  [{idx}/{total_livros}] {nome}: Abreviação não encontrada, pulando...")
            continue
        
        print(f"\n📖 [{idx}/{total_livros}] Importando {nome} ({total_caps} capítulos)...")
        
        for cap in range(1, total_caps + 1):
            total_capitulos += 1
            print(f"   Capítulo {cap}/{total_caps}...", end=" ")
            
            # Buscar texto da API
            texto = buscar_capitulo_api(abreviacao, cap)
            
            if texto:
                # Salvar no banco
                if atualizar_capitulo(livro_id, cap, texto):
                    sucesso += 1
                    print("✅")
                else:
                    falhas += 1
                    print("❌ Erro ao salvar")
            else:
                falhas += 1
                print("❌ Erro ao buscar")
            
            # Aguardar para não sobrecarregar a API
            time.sleep(0.5)
    
    print("\n" + "="*50)
    print(f"✅ Importação concluída!")
    print(f"📊 Total de capítulos processados: {total_capitulos}")
    print(f"✅ Sucessos: {sucesso}")
    print(f"❌ Falhas: {falhas}")
    print("="*50)

if __name__ == "__main__":
    importar_biblia()
